client/src/main/java/uk/co/blackpepper/bowman/LinkedResourceMethodHandler.java

```diff
@@ -1,59 +1,82 @@
 package uk.co.blackpepper.bowman;
 
+import java.beans.PropertyDescriptor;
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
 import java.lang.reflect.ParameterizedType;
 import java.net.URI;
+import java.util.Arrays;
 import java.util.Collection;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.stream.Collectors;
 
 import org.springframework.hateoas.Resource;
 import org.springframework.hateoas.Resources;
 
 import uk.co.blackpepper.bowman.annotation.LinkedResource;
 
-class LinkedResourceMethodHandler implements ConditionalMethodHandler {
-	
+class LinkedResourceMethodHandler extends AbstractContentDelegatingMethodHandler {
+
 	private final RestOperations restOperations;
 
-	private final Resource<?> resource;
-	
 	private final ClientProxyFactory proxyFactory;
-	
+
 	private final PropertyValueFactory propertyValueFactory = new DefaultPropertyValueFactory();
-	
+
 	private final Map<String, Object> linkedResourceResults = new HashMap<>();
-	
+
 	LinkedResourceMethodHandler(Resource<?> resource, RestOperations restOperations, ClientProxyFactory proxyFactory) {
-		this.resource = resource;
+		super(resource);
 		this.restOperations = restOperations;
 		this.proxyFactory = proxyFactory;
 	}
-	
+
 	@Override
 	public boolean supports(Method method) {
 		return method.isAnnotationPresent(LinkedResource.class);
 	}
-	
+
 	@Override
 	public Object invoke(Object self, Method method, Method proceed, Object[] args)
-		throws InvocationTargetException, IllegalAccessException {
-		
-		Object linkedResourceResult = linkedResourceResults.get(method.getName());
-		
-		if (linkedResourceResult == null) {
-			linkedResourceResult = resolveLinkedResource(self, method, proceed, args);
-			linkedResourceResults.put(method.getName(), linkedResourceResult);
+	throws InvocationTargetException, IllegalAccessException {
+		return isSetter(method)
+				? invokeSetter(self, method, proceed, args)
+				: invokeOther(self, method, proceed, args);
+	}
+
+	private Object invokeOther(Object self, Method method, Method proceed, Object[] args)
+	throws InvocationTargetException, IllegalAccessException {
+		if (!linkedResourceResults.containsKey(method.getName())) {
+			linkedResourceResults.put(method.getName(), resolveLinkedResource(self, method, proceed, args));
 		}
-		
-		return linkedResourceResult;
+		return linkedResourceResults.get(method.getName());
+	}
+
+	private Object invokeSetter(Object self, Method method, Method proceed, Object[] args)
+	throws InvocationTargetException, IllegalAccessException {
+		super.invoke(self, method, proceed, args);
```

Comment:
Been mulling this one through, I'm not sure we actually need to call the underlying instance to set the link in the pojo. As long as the handler is called for the getter then it would return the updated link anyway. Just thinking out loud, be interested in your take on it.

Reply:
On a related point - if we merge the `GetterMethodHandler` and `SetterMethodHandler` similarly, there would then be no need for the abstract superclass now the `LinkedResourceMethodHandler` no longer needs it. I wonder if this could then become an `AbstractPropertyAwareMethodHandler` or similar and if this could contain the setter finding code etc., enabling you to DRY this a bit? What do you think?

Reply:
Hi Ryan sorry mate, been snowed under the last few hours. The ```LinkedResourceMethodHandler```, at least at the moment anyway, still requires access to the BeanInfo data as the links are mapped using the non setter method name. Based on your second comment this may not be an issue, assuming I understand your suggestion correctly.

Are you suggesting a ```PropertyAwareMethodHandler``` that is a delegate inside the ```LinkedResourceMethodHandler``` that is delegated to when the method being called is either a getter or setter and that the linked resource is actually stored in the proxied instance variable instead of the map? Then the map in the ```LinkedResourceMethodHandler``` only contains all other non-getter/setter links?

In this scenario the oder of handlers would be ```ResourceIdHandler```, ```LinkedResourceMethodHandler``` finally followed by the same instance of ```PropertyAwareMethodHandler``` passed to the ```LinkedResourceMethodHandler```?

Actually none of that sounds like what you're saying at all :)

Reply:
Actually I think I'm following now. An ```AbstractPropertyAwareMethodHandler``` that has all of the property aware methods and is then extended by a ```PropertyAwareMethodHandler``` and the ```LinkedResourceMethodHandler```?

Reply:
Yes exactly that - only maybe call the concrete impl `PropertyAwareMethodHandler` => `SimplePropertyMethodHandler` or some such - to show it's applicable to all getters and setters of "normal" (i.e. non-linked-resource annotated) properties.

Just a suggestion... what do you think?

Reply:
Yep, pretty much exactly what I have just updated to. Just tidying up the tests now and will update PR soon as complete.